### PR TITLE
fix(TX-446): update order page credit card input

### PR DESCRIPTION
**Pull request by rquartararo**
The type of this PR is: **Fix**

This PR solves [TX-446]

### Description

This PR updates the credit card input to the new version on the orders page and updates the 
border styles to match the other inputs. 

### Screenshot
Screenshot 2022-07-07 at 11 10 44 AM

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TX-446]: https://artsyproduct.atlassian.net/browse/TX-446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Order/Components/CreditCardPicker.tsx [hunk 1]
```diff
@@ -319,14 +319,13 @@ export class CreditCardPicker extends React.Component<
         <Collapse open={this.state.creditCardSelection.type === "new"}>
           {userHasExistingCards && <Spacer mb={2} />}
           <Flex flexDirection="column">
-            <Text mb={1} size="md" color="black100" lineHeight="1.1em">
```
Comment (dzucconi) [inline src/v2/Apps/Order/Components/CreditCardPicker.tsx:322]: Ugh, `md` AND a custom line-height?

**Comment by dzucconi** [comment]
The error you're seeing is because the input accepts `error?: string | boolean` — just like the Palette input.

It's being passed `error={stripeError!}` — which: aside from being incorrectly typed (see below) — is ultimately being passed as `error={{ message: 'string' }}` — which is wrong. Pull the `message` out. But I'd fix all the type errors before doing so.

**Comment by AlicanAkyuz** [comment]
#### video showing different states of card input 
https://user-images.githubusercontent.com/42584148/177966243-b5d1ebb7-2d8a-4000-8b50-7000a0b795bd.mov

File: src/v2/Components/CreditCardInput/CreditCardInput.tsx [hunk 2]
```diff
@@ -99,11 +105,19 @@ const FauxInput = styled(Box)<{ error: boolean }>`
...
 
     &--focus {
+      color: ${themeGet("colors.black100")};
       border-color: ${themeGet("colors.black60")};
+      ::placeholder {
+        color: ${themeGet("colors.black100")};
+      }
     }
 
     &--invalid {
```
Comment (dzucconi) [inline src/v2/Components/CreditCardInput/CreditCardInput.tsx:123]: It doesn't look like this border color changes upon error?

**Review by dzucconi — APPROVED** [review]

File: src/v2/Components/CreditCardInput/CreditCardInput.tsx (hunk 2 above)
Comment (AlicanAkyuz) [reply src/v2/Components/CreditCardInput/CreditCardInput.tsx:123]: think it's because of the hover/focus border color overrides it

**Pull request merged**